lib/mastodon/redis_config.rb

```diff
@@ -42,12 +60,28 @@ def setup_redis_env_url(prefix = nil, defaults = true)
     size: Sidekiq.server? ? Sidekiq[:concurrency] : Integer(ENV['MAX_THREADS'] || 5),
     timeout: 5,
   },
+
+  master_name: (ENV.fetch('CACHE_REDIS_SENTINEL_MASTER', 'mymaster') if ENV['CACHE_REDIS_SENTINEL']),
+  sentinels: (if ENV['CACHE_REDIS_SENTINEL']
+                ENV['CACHE_REDIS_SENTINEL'].split(',').map do |server|
+                  host, port = server.split(':')
+                  { host: host, port: port.to_i }
+                end
+              end),
 }.freeze
 
 REDIS_SIDEKIQ_PARAMS = {
   driver: redis_driver,
   url: ENV['SIDEKIQ_REDIS_URL'],
   namespace: sidekiq_namespace,
+
+  master_name: (ENV.fetch('SIDEKIQ_REDIS_SENTINEL_MASTER', 'mymaster') if ENV['SIDEKIQ_REDIS_SENTINEL']),
+  sentinels: (if ENV['SIDEKIQ_REDIS_SENTINEL']
+                ENV['SIDEKIQ_REDIS_SENTINEL'].split(',').map do |server|
+                  host, port = server.split(':')
+                  { host: host, port: port.to_i }
+                end
+              end),
```

Comment:
This basically looks the same for cache and sidekiq. It is also a lot of logic to cram into what should be a "simple" hash. Maybe this could be moved into a method that is simply called for both cases?